tests/pyccel/test_pyccel.py

```diff
@@ -508,3 +509,63 @@ def test_print_strings(language):
 def test_print_sp_and_end(language):
     types = str
     pyccel_test("scripts/print_sp_and_end.py", language=language, output_dtype=types)
+
+def test_headers(language):
+    test_file = "scripts/test_headers.py"
+    test_file = os.path.normpath(test_file)
+    test_file = get_abs_path(test_file)
+
+    header_file = 'scripts/headers.pyh'
+    header_file = os.path.normpath(header_file)
+    header_file = get_abs_path(header_file)
+
+    with open(test_file, 'w') as f:
+        code = ("from headers import f\n"
+                "def f(x):\n"
+                "    y = x\n"
+                "    return y\n"
+                "print(f(1))\n")
+
+        f.write(code)
+
+    with open(header_file, 'w') as f:
+        code =("#$ header metavar ignore_at_import=True\n"
+               "#$ header function f(int)")
+
+        f.write(code)
+
+    test_file = os.path.normpath(test_file)
+    cwd = os.path.dirname(test_file)
+    cwd = get_abs_path(cwd)
+
+    pyccel_commands = " --language="+language
+
+    compile_pyccel(cwd, test_file, pyccel_commands)
+
+
+    lang_out = get_lang_output(get_exe(test_file))
+    assert int(lang_out)
+
+    with open(test_file, 'w') as f:
+        code = ("from headers import f\n"
+                "def f(x):\n"
+                "    y = x\n"
+                "    return y\n"
+                "print(f(1.))\n")
```

Comment:
Is this the line that lets us check that the mechanism is working?

Reply:
I checked locally with the code:
```
#$ header function f(int)
def f(x):
    y = x
    return y

print(f(1.))
```
I get a compilation error on fortran, but it works for c.

I can get it to fail if you pass: `--flags="-Wconversion -Werror` and pass `1.5` instead of `1.`. Apparently gcc is clever enough to see that that isn't really truncation

Reply:
The same flags are valid for fortran

Reply:
of course if you take an int argument and you pass it a float, it will give you that error, the test doesn't do that

Reply:
I thought that the test did that if the .pyccel file wasn't updated (ie. if it fails). If that isn't the case then can you tell me what exactly tells us that this is working?

Reply:
the fact that we are not getting that error  in github actions means that the test is working

Reply:
so I create a header file and then change it to be sure that the I don't upload the old one

Reply:
>the fact that we are not getting that error in github actions means that the test is working

In that case my point is valid. If the test wasn't working then we would have the code that I printed above. This code raises an error when compiled with pyccel and fortran. But it doesn't raise an error when compiled with pyccel and c. This means that we would think that the test was working when in fact we had the same problem.

If you add the flags that I indicated then the test will still pass, but it will fail for c if there is a problem

Reply:
I'm not testing if it fails to upload or not it always uploads, I'm testing if it uploads the right one after chaning the `-pyh` file

Reply:
yes but I'm never running the code that you printed

Reply:
done